blog-post-review.md

```diff
@@ -0,0 +1,261 @@
+
+Now, let's find the top 3 Pages and visualize their relationships in the graph. We will use the following query in Memgraph Lab:
+```
+MATCH (n)
+WITH n
+ORDER BY n.rank DESC
+LIMIT 3
+MATCH (n)<-[e]-(m)
+RETURN *;
+```
+
+Here in the query first we will `MATCH` all the nodes. In the second part of the query we will `ORDER` nodes by their  `rank` in descending order, and for the first `3` of them get all pages connected to them. We need the `WITH` clause to connect the two parts of the query inside our one query.
+
+![](https://hackmd.io/_uploads/r1mQRhz85.png)
+
+
+Now, that's it considering PageRank, next you will see how to use Louvain community detection to find communities in the graph. 
+
+## Community detection with Louvain 
+The Louvain algorithm measures how connected are the nodes within a community if we would compare them to how connected they would be in a random network. Also, it recursively merges communities into a single node and executes the modularity clustering on the condensed graphs. This is one of the most popular community detection algorithms. Let's run it to find how many communities we have an inside graph. First, we will let Louvain execute and save `cluster_id` as a property for every node. 
+
+
+```python
+memgraph.execute(
+    """
+    CALL cugraph.louvain.get() YIELD cluster_id, node
+    SET node.cluster_id = cluster_id;
+    """
+)
+```
+
+Now, let us find number of communities:
+
+```python
+results =  memgraph.execute_and_fetch(
+        """
+        MATCH (n)
+        WITH DISTINCT n.cluster_id as cluster_id
+        RETURN count(cluster_id ) as num_of_clusters;
+        """
+    )
+# we will get only 1 result
+result = list(results)[0]
+
+#don't forget that results are saved in a dict
+print(f"Number of clusters: {result['num_of_clusters']}")
+
+```
+
+    Number of clusters: 2664
+
+
+Next, you can also visualize some of these communities. You can for example find nodes that belong to one community, but are connected to another node that belongs in the opposing community.
+As for Louvain, it tries to minimize that number of nodes, so we shouldn't see that many of them. In Memgraph Lab try to execute the following query:
+
+```
+MATCH  (n2)<-[e1]-
+WHERE n1.cluster_id != m1.cluster_id AND n1.cluster_id = n2.cluster_id
+RETURN *
+LIMIT 1000;
+```
+Here in our query we will `MATCH`  node `n1` and its two relationships `(n2)<-[e1]-(n1)` and `(n1)-[e]->(m1)` but we will filter out only ones `WHERE` cluster_id of nodes `n1` and `n2` is not the same as the  cluster_id of node `m1`. We will return only 1000 such relationships, for easier visualizations.
```

Comment:
This query will `MATCH` node `n1` and its relationship to (I would explain these relationships a bit more because you say you don't expect your public to know a lot of cypher). Then, it will filter out only those nodes `WHERE` the cluster ID (BCS you are not stating this here as an actual variable, but using English) of `n1` and `n2` nodes is not the same as the cluster ID of node `m1`. Using LIMIT 1000 we limit the results to show only 1000 of such relationships,....